subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/Codecs.kt

```diff
@@ -252,6 +275,33 @@ class Codecs(
         bind(ProviderCodec(nestedCodec))
     }
 
+    /**
+     * Returns a Codec for Provider implementations suitable for all Provider implementations.
+     */
+    private
+    fun nestedProviderCodec(
+        valueSourceProviderFactory: ValueSourceProviderFactory,
+        buildStateRegistry: BuildStateRegistry,
+        flowProviders: FlowProviders
+    ) = FixedValueReplacingProviderCodec(
+        ValueSourceProviderCodec(valueSourceProviderFactory),
+        BuildServiceProviderCodec(buildStateRegistry),
+        FlowProvidersCodec(flowProviders)
+    )
+
+    /**
+     * Returns a Codec for Provider implementations supported in the fingerprinting context. For example, BuildServiceProviders are not supported.
+     */
+    private
+    fun nestedProviderCodecForFingerprint(
+        valueSourceProviderFactory: ValueSourceProviderFactory,
+        flowProviders: FlowProviders
+    ) = FixedValueReplacingProviderCodec(
+        ValueSourceProviderCodec(valueSourceProviderFactory),
+        UnsupportedFingerprintBuildServiceProviderCodec,
+        FlowProvidersCodec(flowProviders)
```

Comment:
I think we should disallow flow providers as well.

Reply:
I don't think you can have `RequestedTasksResultProvider` as part of the fingerprint, as it cannot be isolated at configuration time (yet?), but why not, I can add it just in case

Reply:
Thanks, Mike!